Code/src/main/java/nl/utwente/group10/haskell/env/Env.java

```diff
@@ -1,12 +1,55 @@
 package nl.utwente.group10.haskell.env;
 
 import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
 
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
 import nl.utwente.group10.haskell.type.Type;
+import nl.utwente.group10.haskell.type.TypeClass;
 
 /**
  * Haskell Environment.
  */
-public class Env extends HashMap<String, Type> {
-}
+public class Env {
+    /**
+     * Map containing the types of expressions known to the environment.
```

Comment:
Not all Exprs are contained in the environment (where environment refers to the Env instance), only Idents when properly used.